optuna/visualization/_optimization_history.py

```diff
@@ -74,187 +174,75 @@ def objective(trial):
     """
 
     _imports.check()
-
-    if isinstance(study, Study):
-        studies = [study]
-    else:
-        studies = list(study)
-
-    _check_plot_args(studies, target, target_name)
-    return _get_optimization_history_plot(studies, target, target_name, error_bar)
-
-
-def _get_optimization_history_plot(
-    studies: List[Study],
-    target: Optional[Callable[[FrozenTrial], float]],
-    target_name: str,
-    error_bar: bool,
-) -> "go.Figure":
+    _check_plot_args(study, target, target_name)
 
     layout = go.Layout(
         title="Optimization History Plot",
         xaxis={"title": "Trial"},
         yaxis={"title": target_name},
     )
 
-    if len(studies) == 0:
-        _logger.warning("There are no studies.")
-        return go.Figure(data=[], layout=layout)
-
-    all_trials = list(
-        itertools.chain.from_iterable(
-            study.get_trials(deepcopy=False, states=(TrialState.COMPLETE,)) for study in studies
-        )
-    )
-
-    if len(all_trials) == 0:
-        _logger.warning("There are no complete trials.")
-        return go.Figure(data=[], layout=layout)
-
     if error_bar:
-        return _get_optimization_histories_with_error_bar(studies, target, target_name, layout)
+        info = _get_optimization_history_error_bar_info(study, target, target_name)
+        info_list = None if info is None else [info]
     else:
-        return _get_optimization_histories(studies, target, target_name, layout)
+        info_list = _get_optimization_history_info_list(study, target, target_name)
```

Comment:
This logic should be wrapped and removed from here. I means 
```python
def _get_optimization_history_info_list(...) -> List[_OptimizationHistoryInfo]:
    # Create info list
    
    # If error_bar=False or created info list is dummy, return the info list

    # Create error bar info, and return it


def plot_optimization_history(...) -> go.Figure:
    _imports.check()    
    info_list = _get_optimization_history_info_list(...)
    return _get_optimization_history_plot(info_list)
```

Reply:
Fixed!